.github/workflows/build.yaml

```diff
@@ -1,48 +1,106 @@
-name: build-pipeline
+name: build
 
 on:
-  pull_request:
+  push:
     branches:
       - main
-      - release/v*
+    tags:
+      - v*
 
 jobs:
-  build-frontend:
+  enterprise:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout code
+      - name: checkout
         uses: actions/checkout@v4
-      - name: Install dependencies
-        run: cd frontend && yarn install
-      - name: Build frontend static files
-        shell: bash
-        run: |
-          make build-frontend-static
-
-  build-signoz:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v4
-      - name: Setup golang
+      - name: setup
         uses: actions/setup-go@v4
         with:
           go-version: "1.22"
-      - name: Build signoz image
-        shell: bash
+      - name: setup-qemu
+        uses: docker/setup-qemu-action@v3
+      - name: setup-buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          version: latest
+      - name: docker-login
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: create-env-file
+        run: |
+          echo 'INTERCOM_APP_ID="${{ secrets.INTERCOM_APP_ID }}"' > frontend/.env
+          echo 'SEGMENT_ID="${{ secrets.SEGMENT_ID }}"' >> frontend/.env
+          echo 'SENTRY_AUTH_TOKEN="${{ secrets.SENTRY_AUTH_TOKEN }}"' >> frontend/.env
+          echo 'SENTRY_ORG="${{ secrets.SENTRY_ORG }}"' >> frontend/.env
+          echo 'SENTRY_PROJECT_ID="${{ secrets.SENTRY_PROJECT_ID }}"' >> frontend/.env
+          echo 'SENTRY_DSN="${{ secrets.SENTRY_DSN }}"' >> frontend/.env
+          echo 'TUNNEL_URL="${{ secrets.TUNNEL_URL }}"' >> frontend/.env
+          echo 'TUNNEL_DOMAIN="${{ secrets.TUNNEL_DOMAIN }}"' >> frontend/.env
+          echo 'POSTHOG_KEY="${{ secrets.POSTHOG_KEY }}"' >> frontend/.env
+          echo 'CUSTOMERIO_ID="${{ secrets.CUSTOMERIO_ID }}"' >> frontend/.env
+          echo 'CUSTOMERIO_SITE_ID="${{ secrets.CUSTOMERIO_SITE_ID }}"' >> frontend/.env
+      - uses: benjlevesque/short-sha@v2.2
+        id: short-sha
+      - name: branch-name
+        id: branch-name
+        uses: tj-actions/branch-names@v7.0.7
```

Comment:
Replace `tj-actions/branch-names` with this:

```yaml
      - name: github-ref-info
        shell: bash
        run: |
          GH_REF=${{ github.ref }}
          if [[ "${{ github.ref_type }}" == "tag" ]]; then
            PREFIX="refs/tags/"
            echo "GH_IS_TAG=true" >> $GITHUB_ENV
            echo "GH_TAG=${GH_REF#$PREFIX}" >> $GITHUB_ENV
          else
            PREFIX="refs/heads/"
            echo "GH_IS_TAG=false" >> $GITHUB_ENV
            echo "GH_BRANCH_NAME=${GH_REF#$PREFIX}" >> $GITHUB_ENV
          fi
```

Ref: https://github.com/SigNoz/signoz-otel-collector/blob/main/.github/workflows/push.yaml#L76